crypto/bio/b_sock.c

```diff
@@ -138,6 +164,22 @@ int BIO_get_host_ip(const char *str, unsigned char *ip)
     if (i > 0)
         return (1);
 
+    /* if gethostbyname_r is supported, use it. */
+# ifdef HAVE_GETHOSTBYNAME_R
+    memset(&hostent, 0x00, sizeof(hostent));
+    he = &hostent;
+    err = gethostbyname_r(str, he, buf, sizeof(buf), &result, &h_errnop);
+    if (err != 0 || result == NULL) {
+        BIOerr(BIO_F_BIO_GET_HOST_IP, BIO_R_BAD_HOSTNAME_LOOKUP);
+        /*
+         * if there are no entry found, then gethostbyname_r returns 0 with
+         * result set to NULL.
+         * so reset err to no-zero
+         */
+        err = 1;
+        goto err;
+    }
```

Comment:
If it were me, I would reuse the original error handling and shorten this part substantially:

``` C
    memset(&hostent, 0x00, sizeof(hostent));
    /* gethostbyname_r() sets |he| to NULL on error, we check it further down */
    gethostbyname_r(str, &hostent, buf, sizeof(buf), &he, &h_errnop);
```

Reply:
Note that you can remove `result` entirely, and do feel free to initialize `he` to NULL.

Reply:
sounds good. will do this change.